src/dataflow-types/src/reconciliation/command.rs

```diff
@@ -0,0 +1,361 @@
+                        let iter = frontier.update_iter(changes.drain());
+                        changes.extend(iter);
+                    } else {
+                        changes.clear();
+                    }
+                }
+
+                self.responses.push_back(Response::Compute(
+                    ComputeResponse::FrontierUppers(list),
+                    instance,
+                ));
+            }
+            // Avoid multiple retractions of minimum time, to present as updates from one worker.
+            Response::Storage(StorageResponse::TimestampBindings(mut feedback)) => {
+                for (id, changes) in feedback.changes.iter_mut() {
+                    if let Some(frontier) = self.uppers.get_mut(&(*id, None)) {
+                        let iter = frontier.update_iter(changes.drain());
+                        changes.extend(iter);
+                    } else {
+                        changes.clear();
+                    }
+                }
+
+                self.responses
+                    .push_back(Response::Storage(StorageResponse::TimestampBindings(
+                        feedback,
+                    )));
+            }
+            Response::Compute(ComputeResponse::PeekResponse(uuid, response), instance) => {
+                self.responses.push_back(Response::Compute(
+                    ComputeResponse::PeekResponse(uuid, response),
+                    instance,
+                ));
+            }
+            Response::Compute(ComputeResponse::TailResponse(id, response), instance) => {
+                self.responses.push_back(Response::Compute(
+                    ComputeResponse::TailResponse(id, response),
+                    instance,
+                ));
+            }
+        }
+    }
+
+    async fn observe_compute_command(
+        &mut self,
+        command: ComputeCommand,
+        instance: ComputeInstanceId,
+    ) -> Result<(), anyhow::Error> {
+        use Command::Compute;
+        use ComputeCommand::*;
+        match command {
+            CreateInstance(config, logging) => {
+                if !self.created {
+                    self.created = true;
+                    if let Some(logging) = &logging {
+                        for id in logging.log_identifiers() {
+                            if !self.uppers.contains_key(&(id, Some(instance))) {
+                                self.start_tracking(id, Some(instance));
+                            }
+                        }
+                    }
+                    self.client
+                        .send(Compute(CreateInstance(config, logging), instance))
+                        .await?;
+                }
+                Ok(())
+            }
+            cmd @ DropInstance => {
+                if self.created {
+                    self.created = false;
+                    self.uppers.clear();
+                    self.client.send(Compute(cmd, instance)).await
+                } else {
```

Comment:
This is where I became confused about `self.created` not consulting the instance id before trashing the whole place. :D